aria_data_loaders/aria_data_utils/aria_live_streamer.py

```diff
@@ -0,0 +1,626 @@
+        self._sensors_calib_json = None
+        self._sensors_calib: Optional[aria_core.calibration.SensorCalibration] = None
+        self._device_T_cpf = None
+        self._rgb_calib_params: Optional[aria_core.calibration.CameraCalibration] = None
+        self._dst_calib_params: Optional[aria_core.calibration.CameraCalibration] = None
+        self.device_T_camera: Optional[sp.SE3] = None
+        self.device, self.device_client = self._setup_aria()
+
+        self.static_tf_broadcaster = StaticTransformBroadcaster()
+
+        # Maintain a list of all poses where qr code is detected (w.r.t ariaWorld)
+        self.marker_positions_list = (
+            []
+        )  # type: List[np.ndarray] # List of  position as np.ndarray (x, y, z)
+        self.marker_quaternion_list = (
+            []
+        )  # type: List[np.ndarray] # List of quaternions as np.ndarray (x, y, z, w)
+        self.fix_list_length = 10  # TODO: Add this to config
+        self.avg_ariaWorld_T_marker = None  # type: Optional[sp.SE3]
+
+    def connect(self):
+        """
+        Connect to Aria
+        """
+        self.device.streaming_manager.streaming_client.subscribe()  # type: ignore
+        self._is_connected = True
+        rospy.loginfo("Connected to Aria")
+
+    def disconnect(self):
+        """
+        Unsubscribes from Aria's streaming client & disconnects from Aria
+        """
+        self._is_connected = False
+        self.device.streaming_manager.streaming_client.unsubscribe()  # type: ignore
+        self.device_client.disconnect(self.device)  # type: ignore
+
+    def _setup_aria(self) -> Tuple[aria.Device, aria.DeviceClient]:
+        """
+        Setup aria device and device client with appropriate configurations for live streaming
+
+        Updates:
+            - self._sensors_calib_json - JSON string of sensors calibration
+            - self._sensors_calib - aria's SensorCalibration object
+            - self._rgb_calib_params - aria's CameraCalibration params object for camera-rgb
+            - self._dst_calib_params - desired CameraCalibration params with linear distortion model
+            - self.device_T_camera - Sophus SE3 transform from device frame (left-slam camera) to rgb-camera frame
+
+        Returns:
+            Tuple[aria.Device, aria.DeviceClient]: Tuple of aria's device and device client objects after their proper setup
+        """
+        # get a device client and configure it
+        device_client = aria.DeviceClient()
+        client_config = aria.DeviceClientConfig()
+        device_client.set_client_config(client_config)
+
+        # create a device
+        device = device_client.connect()
+
+        # get the path to this file and load the static device_T_cpf matrix
+        this_file_path = os.path.dirname(os.path.realpath(__file__))
+        self._device_T_cpf = np_matrix3x4_to_sophus_SE3(
+            np_matrix=np.load(
+                os.path.join(this_file_path, "../device_model/device_T_cpf.npy")
```

Comment:
where do you get device_T_cpf.npy?

Reply:
device_T_cpf transformation is a constant transform.
Device frame is the Left Slam Camera, CPF is a frame at center of aria.

More info on frames can be found here - https://facebookresearch.github.io/projectaria_tools/docs/data_formats/coordinate_convention/3d_coordinate_frame_convention

Reply:
added the link to that line in code. @rutadesai